### PR TITLE
Remove MapperService dependency from DocTableInfo.writeTo

### PR DESCRIPTION
**Pull request by mfussenegger**
Relates to https://github.com/crate/crate/issues/11939

### CODE REVIEW

File: server/src/main/java/io/crate/metadata/doc/DocTableInfo.java [hunk 1]
```diff
@@ -968,11 +965,14 @@ public Metadata.Builder writeTo(CheckedFunction<IndexMetadata, MapperService, IO
             if (indexMetadata == null) {
                 throw new UnsupportedOperationException("Cannot create index via DocTableInfo.writeTo");
             }
-            MapperService mapperService = createMapperService.apply(indexMetadata);
-            DocumentMapper mapper = mapperService.merge(mapping, MapperService.MergeReason.MAPPING_UPDATE);
+
+            long allowedTotalColumns = MapperService.INDEX_MAPPING_TOTAL_FIELDS_LIMIT_SETTING.get(indexMetadata.getSettings());
```
Comment (romseygeek) [inline server/src/main/java/io/crate/metadata/doc/DocTableInfo.java:969]: Is it worth moving this setting into DocTableInfo?

File: server/src/test/java/io/crate/integrationtests/DDLIntegrationTest.java [hunk 2]
```diff
@@ -1005,7 +1024,7 @@ public void test_add_geo_shape_column() throws Exception {
                    "id" INTEGER,
                    "g" GEO_SHAPE GENERATED ALWAYS AS 'POLYGON (( 5 5, 30 5, 30 30, 5 30, 5 5 ))' INDEX USING QUADTREE WITH (
                       distance_error_pct = 0.123,
-                      precision = '123.0m'
+                      precision = '123m'
```
Comment (romseygeek) [inline server/src/test/java/io/crate/integrationtests/DDLIntegrationTest.java:1027]: Just out of interest, what caused this change?

**Review by romseygeek — APPROVED** [review]
Nice.

I was wondering if we could also remove IndicesService.createIndexMapperService() as part of this, but that's still used for alter table requests that update settings.  It probably doesn't need to be though, so that can be a nice follow-up.

File: server/src/main/java/io/crate/metadata/doc/DocTableInfo.java (hunk 1 above)
Comment (mfussenegger) [reply server/src/main/java/io/crate/metadata/doc/DocTableInfo.java:969]: Eventually yes. My plan was to move it once the existing uses within the MapperService are also gone

File: server/src/test/java/io/crate/integrationtests/DDLIntegrationTest.java (hunk 2 above)
Comment (mfussenegger) [reply server/src/test/java/io/crate/integrationtests/DDLIntegrationTest.java:1027]: I haven't double checked but I suspect the geo field mappers normalized the representation

**Comment by mfussenegger** [comment]
> I was wondering if we could also remove IndicesService.createIndexMapperService() as part of this, but that's still used for alter table requests that update settings. It probably doesn't need to be though, so that can be a nice follow-up.

Good point. I've some work for alter table already queued up, but need to shuffle around some more things before I'll get back to that.

**Pull request merged**